services/mail.js

```diff
@@ -62,6 +62,34 @@ function generateHtmlEmail({ template, templateData }) {
     });
 }
 
+/**
+ * Utility function to generate an HTML email and send it.
+ *
+ *
+ * @param {object} options
+ * @param {string} options.template
+ * @param {object} options.templateData
+ * @param {object} mailParams
+ * @return {object}
+ */
+async function sendHtmlEmail({ template, templateData }, mailParams) {
```

Comment:
Nice, could you add a test for this? It might be more a case of reworking some of the current ones to test this higher-level function instead of the constituent parts.

Reply:
Done – give it a look now.